src/ui/Button.cpp

```diff
@@ -0,0 +1,204 @@
+﻿// Button.cpp
+
+#include "../main.hpp"
+#include "../player.hpp"
+#include "../draw.hpp"
+#include "Frame.hpp"
+#include "Button.hpp"
+#include "Image.hpp"
+#include "Text.hpp"
+
+Button::Button() {
+	size.x = 0; size.w = 32;
+	size.y = 0; size.h = 32;
+	color = SDL_MapRGBA(mainsurface->format, 127, 127, 127, 255);
+	textColor = SDL_MapRGBA(mainsurface->format, 255, 255, 255, 255);
+}
+
+Button::Button(Frame& _parent) : Button() {
+	parent = &_parent;
+	_parent.getButtons().push_back(this);
+	_parent.adoptWidget(*this);
+}
+
+Button::~Button() {
+	if (callback) {
+		delete callback;
+		callback = nullptr;
+	}
```

Comment:
How is cleanup initiated? No message needed to the parent; does the parent initiate the destruction of every button?